Source/Project64/N64 System/Mips/Pif Ram.cpp

```diff
@@ -297,19 +297,16 @@ void CPifRam::SI_DMA_READ()
 			{
 				continue;
 			}
-			RDRAM[RdramPos ^3] = m_PifRam[count];
+			RDRAM[BES(RdramPos)] = m_PifRam[count];
 		}
 	}
 	else
 	{
-		for (size_t i = 0; i < 64; i += 4)
-		{
-			unsigned __int32 pif_ram_dword;
-			std::memcpy(&pif_ram_dword, &PifRamPos[i], sizeof(unsigned __int32));
-
-			pif_ram_dword = swap32by8(pif_ram_dword);
+		register size_t i;

```

Comment:
In closing, I'd like to address some things.

> Would you, now? I almost find that impressive.

You presented me with a, 'between a rock and a hard place' scenario. While I would not agree with a standard decision on that from a design standpoint., it's more a matter of, "In [x] years, will my program still be considered valid code on newer standards?". Compatibility works both ways. Moving to a new language is a consideration, but at that point that begs the questions of whether or not that's possible (assuming constraints, etc), or worth the time.

> And always have existed--just as they have in C89. However, whether the loop heading (as opposed to the body) constitutes its own existing "scope" is again a matter of one's own (and the individual language's) personal interpretation. Maybe that could be another reason why ANSI C does not support what you say.

You can likely get a definitive answer to this by e-mailing someone related to the ISO C Working Group (there are contact links on the working group's site).

> Good theory but I have an even easier one: Respect backwards compatibility.

The same can be said of forward compatibility as well.

That's all I wanted to address in particular.